frontend/src/components/GameMap/Chapter.js

```diff
@@ -0,0 +1,51 @@
+import React, { useRef } from 'react';
+import { Border, MyContainer } from './ChapterStyles';
+import { Row } from 'react-bootstrap';
+import Activity from './Activity';
+
+
+function getActivityFromMap(map, x, y)
+{
+    for (const activity of map.activities)
+    {
+        if (activity.posX === x && activity.posY === y) 
+            return activity;
+    }
+    return null;
```

Comment:
A można by to zastąpić przez:
```
return map.activities.find(activity => activity.posX === x && activity.posY === y) || null
```
?
Find działa tak że szuka w podanym array elementu spełniającego podany warunek. Jeżeli znajdzie taki element to go zwraca (pierwszy jaki znajdzie), a jeżeli nie znajdzie to zwraca undefined. Stąd można dać dalej `|| null` i wtedy jak nic nie znajdzie to zwróci `null`